<launch>
  <!-- Run Val IK Services-->
  <node name="val_ik" pkg="val_ik" type="val_ik_service"/>
  <node name="val_ik_single_ik_server_node" pkg="val_ik" type="val_single_ik_srv"/>       

  <!-- Run Interactive Markers -->
  <node name="basic_controls" pkg="val_logic_manager" type="interactive_marker"/>

  <!-- Run Val Logic-->
  <node name="val_logic_manager" pkg="val_logic_manager" type="logic_main" output="screen"/>

  <!-- Run Python GUI-->
  <node name="steven_simple_val_gui" pkg="val_logic_manager" type="GUI_val_logic.py" output="screen"/>

  <!-- Publish Sole Frames -->
  <include file="$(find val_logic_manager)/launch/soleFrame_static_tf.launch"/>

  <!-- Run Self Filter and Object Registration Nodes -->
  <include file="$(find val_logic_manager)/launch/self_filter_and_object_registration_nodes.launch"/>

  <!-- Run RVIZ -->
  <include file="$(find val_logic_manager)/launch/val_ik_viz.launch"/>

</launch>